diff -rupN a/libavutil/mem.h b/libavutil/mem.h
--- a/libavutil/mem.h	2018-11-18 03:58:00.000000000 +0300
+++ b/libavutil/mem.h	2019-07-06 19:10:54.000000000 +0300
@@ -338,7 +338,7 @@ av_alloc_size(2, 3) void *av_realloc_arr
  * @warning Unlike av_malloc(), the allocated memory is not guaranteed to be
  *          correctly aligned.
  */
-av_alloc_size(2, 3) int av_reallocp_array(void *ptr, size_t nmemb, size_t size);
+int av_reallocp_array(void *ptr, size_t nmemb, size_t size);
 
 /**
  * Reallocate the given buffer if it is not large enough, otherwise do nothing.
