--- a/src/video/android/SDL_androidevents.c		2022-02-03 17:01:24.232333800 +1100
+++ b/src/video/android/SDL_androidevents.c		2022-02-03 17:04:44.719725000 +1100
@@ -116,7 +116,7 @@ Android_PumpEvents_Blocking(_THIS)
         /* Make sure this is the last thing we do before pausing */
         if (!isContextExternal) {
             SDL_LockMutex(Android_ActivityMutex);
-            android_egl_context_backup(Android_Window);
+            //android_egl_context_backup(Android_Window);
             SDL_UnlockMutex(Android_ActivityMutex);
         }
 
@@ -140,7 +140,7 @@ Android_PumpEvents_Blocking(_THIS)
             /* Restore the GL Context from here, as this operation is thread dependent */
             if (!isContextExternal && !SDL_HasEvent(SDL_QUIT)) {
                 SDL_LockMutex(Android_ActivityMutex);
-                android_egl_context_restore(Android_Window);
+                //android_egl_context_restore(Android_Window);
                 SDL_UnlockMutex(Android_ActivityMutex);
             }
 
@@ -190,7 +190,7 @@ Android_PumpEvents_NonBlocking(_THIS)
 
             if (!isContextExternal) {
                 SDL_LockMutex(Android_ActivityMutex);
-                android_egl_context_backup(Android_Window);
+                //android_egl_context_backup(Android_Window);
                 SDL_UnlockMutex(Android_ActivityMutex);
             }
 
@@ -222,7 +222,7 @@ Android_PumpEvents_NonBlocking(_THIS)
             /* Restore the GL Context from here, as this operation is thread dependent */
             if (!isContextExternal && !SDL_HasEvent(SDL_QUIT)) {
                 SDL_LockMutex(Android_ActivityMutex);
-                android_egl_context_restore(Android_Window);
+                //android_egl_context_restore(Android_Window);
                 SDL_UnlockMutex(Android_ActivityMutex);
             }
 
