diff -rupN a/configure.in b/configure.in
--- a/configure.in	2017-11-04 01:13:07.000000000 +0300
+++ b/configure.in	2017-11-04 02:05:34.000000000 +0300
@@ -3559,7 +3559,7 @@ AC_HELP_STRING([--enable-render-d3d], [e
         VERSION_SOURCES="$srcdir/src/main/windows/*.rc"
         SDLMAIN_SOURCES="$srcdir/src/main/windows/*.c"
         SDL_CFLAGS="$SDL_CFLAGS -Dmain=SDL_main"
-        SDL_LIBS="-lSDL2main $SDL_LIBS -mwindows"
+        SDL_LIBS="-lSDL2main $SDL_LIBS -mwindows -lshlwapi"
 
         # Check to see if this is a mingw or cygwin build
         have_mingw32=
diff -rupN a/src/core/windows/SDL_xinput.c b/src/core/windows/SDL_xinput.c
--- a/src/core/windows/SDL_xinput.c	2017-10-23 22:27:46.000000000 +0300
+++ b/src/core/windows/SDL_xinput.c	2017-11-04 02:08:36.000000000 +0300
@@ -23,6 +23,7 @@
 #include "SDL_assert.h"
 #include "SDL_xinput.h"
 
+#include <shlwapi.h>
 
 #ifdef HAVE_XINPUT_H
 
@@ -75,6 +76,8 @@ int
 WIN_LoadXInputDLL(void)
 {
     DWORD version = 0;
+    WCHAR basename[MAX_PATH];
+    WCHAR emuLibrary[MAX_PATH];
 
     if (s_pXInputDLL) {
         SDL_assert(s_XInputDLLRefCount > 0);
@@ -82,8 +85,16 @@ WIN_LoadXInputDLL(void)
         return 0;  /* already loaded */
     }
 
-    version = (1 << 16) | 4;
-    s_pXInputDLL = LoadLibrary(L"XInput1_4.dll");  /* 1.4 Ships with Windows 8. */
+    GetModuleFileNameW(NULL, basename, MAX_PATH);
+    PathRemoveFileSpecW(basename);
+    wsprintf(emuLibrary, L"%ls\\XInput1_3.dll", basename);
+
+    version = (1 << 16) | 3;
+    s_pXInputDLL = LoadLibrary( emuLibrary ); /* For various emulators put into the application folder */
+    if (!s_pXInputDLL) {
+        version = (1 << 16) | 4;
+        s_pXInputDLL = LoadLibrary( L"XInput1_4.dll" );  /* 1.4 Ships with Windows 8. */
+    }
     if (!s_pXInputDLL) {
         version = (1 << 16) | 3;
         s_pXInputDLL = LoadLibrary(L"XInput1_3.dll");  /* 1.3 can be installed as a redistributable component. */
